ui/lib/apps/SlowQuery/components/SlowQueriesTable.tsx

```diff
@@ -53,49 +29,22 @@ function tableColumns(
 interface Props extends Partial<ICardTableV2Props> {
   loading: boolean
   slowQueries: SlowqueryBase[]
-
-  orderBy: string
-  desc: boolean
   showFullSQL?: boolean
-
-  onChangeSort: (orderBy: string, desc: boolean) => void
-  onGetColumns?: (columns: IColumn[]) => void
 }
 
 export default function SlowQueriesTable({
   loading,
   slowQueries,
-
-  orderBy,
-  desc,
   showFullSQL,
 
-  onChangeSort,
-  onGetColumns,
   ...restProps
 }: Props) {
   const navigate = useNavigate()
 
-  const columns = tableColumns(
+  const columns = useMemo(() => tableColumns(slowQueries, showFullSQL), [
```

Comment:
I suspect this violates the react hook principle since `tableColumns` is in fact a hook (or let's say, it contains hooks), and `useMemo()` causes hook calls being conditional

Reply:
I doubt it as well, and in fact, it is true that we can't use `tableColumns()` inside the `useMemo()`, `useState(()=> {...})`, `useEffect()` before, else it will crash and warns us we can't use hooks inside them, that's because we call the `useTranslations()` directly inside a `useSuccessColumn()` method to make it become the hook, after moving the `useTranslation()` to the `ResultStatusBadge` function component, the `useSuccessColumn()` isn't the hook any more.

And although the name field returns a component `TextWithInfo.TransKey` which uses `useTranslations()` inside, but it won't make the `useXxxColumn()` method become the hook, this refreshed my understanding.

Reply:
Cool! The idea is brilliant. I suggest to remove the `useXxx` prefix to indicate that they are no longer hooks.

Reply:
Agree, will update it soon!

Reply:
There are too many `useXxxColumn()` methods, so I will merge this PR first, will tidy all `useXxxColumn()` in the next PR.